tests/unit/lib/build_module/test_workflow_config.py

```diff
@@ -70,6 +70,30 @@ def test_must_work_for_java(self, runtime, build_file, dep_manager, os_mock):
         else:
             self.assertIsNone(result.executable_search_paths)
 
+    @parameterized.expand([
+        ("provided", "requirements.txt", "pip", "python"),
+        ("provided", "package.json", "npm", "nodejs"),
+        ("provided", "Gemfile", "bundler", "ruby"),
+        ("provided", "build.gradle", "gradle", "java"),
+        ("provided", "build.gradle.kts", "gradle", "java"),
+        ("provided", "pom.xml", "maven", "java")
+    ])
+    @patch("samcli.lib.build.workflow_config.os")
+    def test_must_work_for_provided_runtime(self, runtime, build_file, dep_manager, language, os_mock):
+        os_mock.path.join.side_effect = lambda dirname, v: v
+        os_mock.path.exists.side_effect = lambda v: v == build_file
+
+        result = get_workflow_config(runtime, self.code_dir, self.project_dir)
+        self.assertEquals(result.language, language)
+        self.assertEquals(result.dependency_manager, dep_manager)
+        self.assertEquals(result.application_framework, None)
+        self.assertEquals(result.manifest_name, build_file)
+
+        if dep_manager == "gradle":
+            self.assertEquals(result.executable_search_paths, [self.code_dir, self.project_dir])
+        else:
+            self.assertIsNone(result.executable_search_paths)
+
```

Comment:
@sanathkr Would this be sufficient?

Reply:
fair enough, the appropriate builder gets selected. A integration test for this, would be great as well.